driver/src/main/com/mongodb/Mongo.java

```diff
@@ -714,70 +711,49 @@ private static Cluster createCluster(final MongoClientURI mongoURI, final MongoD
-                                                  new SocketStreamFactory(options.getHeartbeatSocketSettings(),
-                                                                          options.getSslSettings(),
-                                                                          options.getSocketFactory()),
-                                                  credentialsList, null,
-                                                  new JMXConnectionPoolListener(), null,
-                                                  createCommandListener(options.getCommandListeners()),
-                                                  options.getApplicationName(),
-                                                  mongoDriverInformation);
-    }
-
-    private static CommandListener createCommandListener(final List<CommandListener> commandListeners) {
-        switch (commandListeners.size()) {
-            case 0:
-                return null;
-            case 1:
-                return commandListeners.get(0);
-            default:
-                return new CommandEventMulticaster(commandListeners);
-        }
+        return createCluster(getClusterSettings(singletonList(serverAddress), options, getSingleServerClusterMode(options)),
+                credentialsList, options, mongoDriverInformation);
     }
 
-    private static List<ServerAddress> createNewSeedList(final List<ServerAddress> seedList) {
-        List<ServerAddress> retVal = new ArrayList<ServerAddress>(seedList.size());
-        for (final ServerAddress cur : seedList) {
-            retVal.add(cur);
+    private static Cluster createCluster(final ClusterSettings clusterSettings, final List<MongoCredential> credentialsList,
+                                         final MongoClientOptions options, final MongoDriverInformation mongoDriverInformation) {
+        return new DefaultClusterFactory().createCluster(clusterSettings,
+                options.getServerSettings(),
+                options.getConnectionPoolSettings(),
+                new SocketStreamFactory(options.getSocketSettings(),
+                                        options.getSslSettings(),
+                                        options.getSocketFactory()),
+                new SocketStreamFactory(options.getHeartbeatSocketSettings(),
+                                        options.getSslSettings(),
+                                        options.getSocketFactory()),
+                credentialsList,
+                getCommandListener(options.getCommandListeners()),
+                options.getApplicationName(),
+                mongoDriverInformation);
+    }
+
+    private static ClusterSettings getClusterSettings(final List<ServerAddress> seedList, final MongoClientOptions options,
+                                                      final ClusterConnectionMode clusterConnectionMode) {
+        List<ServerAddress> hosts = new ArrayList<ServerAddress>();
+        hosts.addAll(seedList);
```

Comment:
Nit: this can be simplified further by just passing seedList to the ArrayList constructor that takes a Collection instead of calling addAll.  Then the hosts variable itself can be inlined.